src/plugins/discover/server/saved_objects/search_migrations.ts

```diff
@@ -120,9 +124,45 @@ const migrateSearchSortToNestedArray: SavedObjectMigrationFn<any, any> = (doc) =
   };
 };
 
+/**
+ * This creates a migration map that applies search source migrations
+ */
+const getSearchSourceMigrations = (searchSourceMigrations: MigrateFunctionsObject) =>
+  mapValues<MigrateFunctionsObject, MigrateFunction>(
+    searchSourceMigrations,
+    (migrate: MigrateFunction<SerializedSearchSourceFields>): MigrateFunction =>
+      (state) => {
+        const _state = state as unknown as { attributes: SavedSearchAttributes };
+
+        const parsedSearchSourceJSON = _state.attributes.kibanaSavedObjectMeta.searchSourceJSON;
+
+        if (!parsedSearchSourceJSON) return _state;
+
+        return {
```

Comment:
@pgayvallet applied the change of types because else I get an error here:
<img width="1634" alt="Bildschirmfoto 2022-03-17 um 09 28 01" src="https://user-images.githubusercontent.com/463851/158769059-5e058609-e00a-4239-bea6-fe5cda84d517.png">

Reply:
so the other way could be to fix it here